ui/ui-config/pages/app/new/index.tsx

```diff
@@ -20,87 +24,96 @@ class CustomErrorHandler extends EventDispatcher<{ message: string }> {
       this.message = message;
 
       if (this.message?.config?.url === "/api/v1/status/public") {
-        console.info(`could not reach API: ${message}`);
+        console.info(`Could not reach API: ${message}`);
       } else {
         errorHandler.set({ message: this.message });
       }
     });
   }
 }
 
-export default function Page(props) {
-  const {
-    plugins,
-  }: {
-    plugins: Actions.PluginsList["plugins"];
-  } = props;
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
 
+export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
+  const client = generateClient(ctx);
+  const { plugins } = await client.request<Actions.PluginsList>(
+    "get",
+    `/plugins`,
+    {
+      includeInstalled: true,
+      includeAvailable: true,
+      includeVersions: false,
+    }
+  );
+  return { props: { plugins } };
+};
+
+const Page: NextPageWithInferredProps<typeof getServerSideProps> = ({
+  plugins,
+}) => {
   const router = useRouter();
   const { client } = useApi();
-  const [plugin, setPlugin] = useState<
-    Partial<Actions.PluginsList["plugins"][number]>
-  >({ name: "" });
-  const [cards, setCards] = useState([]);
+  const [plugin, setPlugin] = useState<Partial<PluginWithVersion>>(() => ({
+    name: "",
+  }));
```

Comment:
I don't precisely recall how the computed state works, but since you're setting this to a static object I don't think you need to do so.

Reply:
I'm just cleaning the syntax here. This file needed more work but wanted to focus on fixing the bug.